benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +43,190 @@ def get_data_model_from_task(task, param_dict):
                     f"Unsupported data_model '{data_model}' defined for task '{task}'"
                 )
     return None
+
+
+def _partition_input_files(input_files, task_options):
+    """
+    This function partitions the input files into witness and other files.
+    The distinction is based on the file name of the witness file, which
+    is identified by the option "witness" in the task options.
+
+    @param input_files: List of input files
+    @param task_options: Dictionary of task options
+    @return: Tuple of witness files and other files
+    """
+    witness_files = []
+    other_files = []
+    for file in input_files:
+        if Path(file).name == task_options.get("witness"):
```

Comment:
Yes, and the reason for this is also the reason for why there even is this duplication at all.

The problem is that the witness is a relative path, and needs to be treated like all other input paths, i.e., BenchExec needs to adjust it to make it a path that is relative to the working directory of the tool. But all the stuff inside the `options` keyword is (from BenchExec's point of view) just arbitrary additional information that it passes through to the tool-info module. We do not want to hard-code information about specific keys in this dict. (If we would for example require in BenchExec that `witness` is a string, this would prevent other communities from using the same key for something else.) So we decided to list the witness as an input file as well. For them, BenchExec knows that it needs to do the proper path adjustment. So the tool-info module gets all the correct relative paths, and needs to filter out the one for the witness. For this only the base name of the witness file is usable.